.pre-commit.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env sh
+# pre-commit.sh
+
+STASH_NAME="pre-commit-$(date +%s)"
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m'
+
+if git stash save -u -k -q $STASH_NAME; then
+    echo "${YELLOW}Stashed changes as:${NC} ${STASH_NAME}\n\n"
+    stash=1
```

Comment:
I tried setting `stash` to 0 at the beginning and it seemed to change the behavior; now, if i test the script on a commit with no staged changes and a clean working directory, the script stashes the changes and then when it tries to un-stash it fails and results in unmerged paths, requiring me to `git reset HEAD -- && git checkout .` to get back to my pre-commit state.